DataformatTest.snakeYaml test throws NoClassDefFoundError trustedTagInspector

### Bug description

Quarkus 3.4.0 upgraded SnakeYAML to 2.1 and we see:

```
Caused by: java.lang.NoClassDefFoundError: org/yaml/snakeyaml/inspector/TrustedTagInspector
 	at org.apache.camel.component.snakeyaml.SnakeYAMLDataFormat.getYaml(SnakeYAMLDataFormat.java:146)
 	at org.apache.camel.component.snakeyaml.SnakeYAMLDataFormat.marshal(SnakeYAMLDataFormat.java:115)
 	at org.apache.camel.support.processor.MarshalProcessor.process(MarshalProcessor.java:64)
 	... 47 more
```

--- a/integration-tests/dataformat/pom.xml
+++ b/integration-tests/dataformat/pom.xml
@@ -67,7 +67,6 @@
 
 
     <profiles>
-        <!-- TODO: https://github.com/apache/camel-quarkus/issues/5278
         <profile>
             <id>native</id>
             <activation>
@@ -95,7 +94,6 @@
                 </plugins>
             </build>
         </profile>
-        -->
         <profile>
             <id>virtualDependencies</id>
             <activation>

--- a/integration-tests/dataformat/src/test/java/org/apache/camel/quarkus/component/dataformat/it/DataformatTest.java
+++ b/integration-tests/dataformat/src/test/java/org/apache/camel/quarkus/component/dataformat/it/DataformatTest.java
@@ -29,7 +29,6 @@ import io.quarkus.test.junit.QuarkusTest;
 import io.restassured.RestAssured;
 import org.apache.commons.io.IOUtils;
 import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.MethodSource;
@@ -43,7 +42,6 @@ class DataformatTest {
         return Stream.of("dataformat-component", "dsl");
     }
 
-    @Disabled("https://github.com/apache/camel-quarkus/issues/5278")
     @ParameterizedTest
     @MethodSource("snakeyamlRoutes")
     public void snakeYaml(String route) {

